/*
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

 This patch throttles jsplumb mouse move events to 50 FPS

diff --git a/silk-workbench/silk-workbench-core/public/libs/jsplumb/dist/js/jsplumb.js b/silk-workbench/silk-workbench-core/public/libs/jsplumb/dist/js/jsplumb.js
--- a/silk-workbench/silk-workbench-core/public/libs/jsplumb/dist/js/jsplumb.js
+++ b/silk-workbench/silk-workbench-core/public/libs/jsplumb/dist/js/jsplumb.js
@@ -1651,7 +1651,7 @@
             }
         }.bind(this);
 
-        this.moveListener = function(e) {
+        var moveListener = function(e) {
             if (downAt) {
                 if (!moving) {
                     var _continue = _dispatch("start", {el:this.el, pos:posAtDown, e:e, drag:this});
@@ -1677,6 +1677,8 @@
             }
         }.bind(this);
 
+        this.moveListener = _.throttle(moveListener, 20);
+
         this.upListener = function(e) {
             if (downAt) {
                 downAt = null;
